pyneuroml/plot/Plot.py

```diff
@@ -240,6 +249,41 @@ def generate_plot(
         logger.info("Saved image to %s of plot: %s" % (save_figure_to, title))
 
     if show_plot_already:
+        if animate:
+            d = 5
+            duration = d * 1000 # in ms
+            size = len(xvalues[0])
+            interval = 50  # Delay between frames in milliseconds
+            pockets = duration // interval
+            skip = max(size // pockets, 1)
+            logger.info(
+                "Animation hyperparameters : duration=%s, size=%s, interval=%s, pockets=%s, skip=%s" % (duration, size, interval, pockets,  skip))
+
+            def update(frame):
+                for i, artist in enumerate(artists):
+                    artist.set_xdata(xvalues[i][:frame*skip])
+                    artist.set_ydata(yvalues[i][:frame*skip])
+                return artists
+            
+            ani = animation.FuncAnimation(
+                fig=fig,
+                frames=size-1,
+                func=update,
+                interval=interval,
+                blit=True,
+                cache_frame_data=False
+            )
+
+            frame_length_threshold = 5000
+            if len(xvalues[0]) < frame_length_threshold and save_animation_to:
```

Comment:
This is to limit animations to 5000 frames only? If this depends on the users' setups (how good their hardware etc. is), let's not add this check like this. Maybe we can just add a warning that is thrown if there are more than 5000 points that says "too many data points, this may take a while"? That way the user can still go ahead and use it if they wish. Maybe they'll start it and go eat lunch, and it'll be complete by the time they return (we do this a lot with our simulations :))

Reply:
I see. I will add a warning and it will ask for user input to continue with the saving or not (possibly showing approx time (minutes) it might take to complete the entire process. 
It will then save or discard based on the user input. 
How does that sound ?

Reply:
I'd skip the user input entirely. Just make it an argument and if it's more than something, just print a warning saying "this could take a while". If it takes too long, the user can always interrupt it and use a smaller value.

(Adding interaction to take user input will make it trickier to do batch analyses, so we want to avoid that as much as possible)